connection.go

```diff
@@ -721,6 +728,18 @@ func (c *Connection) writeFrames(_ uint32) {
 	}
 }
 
+// updateLastActivity marks when the last message was received/sent on the channel.
+// This is used for monitoring idle connections and timing them out.
+func (c *Connection) updateLastActivity(frame *Frame) {
+	// Pings are ignored for last activity.
+	switch frame.Header.messageType {
+	case messageTypeCallReq, messageTypeCallReqContinue, messageTypeCallRes, messageTypeCallResContinue, messageTypeError:
+		c.stateMut.Lock()
```

Comment:
little worried about lock overhead here, we could consider an `atomic.Int64` and store `t.UnixNano()` which gives us enough precision and should be cheaper (a single atomic operation vs the cheapest for a lock which is two atomic ops)

Reply:
Done